src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -6,6 +6,38 @@ namespace System.Collections.Tests
 {
     public abstract partial class Dictionary_Generic_Tests<TKey, TValue> : IDictionary_Generic_Tests<TKey, TValue>
     {
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_TryAdd(int count)
+        {
+            var dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+
+            TKey key = CreateTKey(seed: count);
+            TValue value = dictionary.ContainsKey(key) ? dictionary[key] : default(TValue);
+            int originalCount = dictionary.Count;
+            bool alreadyContainsKey = dictionary.ContainsKey(key);
+
+            Assert.Equal(!alreadyContainsKey, dictionary.TryAdd(key, default(TValue)));
+            Assert.Equal(alreadyContainsKey ? originalCount : originalCount + 1, dictionary.Count);
+            Assert.Equal(dictionary[key], value);
+
+            if (!dictionary.Any())
+            {
+                return;
+            }
+
+            // It's unlikely that randomly generated data will already be in the dictionary,
```

Comment:
I don't understand this comment... we already added key to the dictionary, so don't we know it's in there?